### ISSUE TITLE
How to do Footnotes/Citations

### ISSUE THREAD

**Issue by jlivingood** [bitag-twg/bitag2.0#17]
How do we want to do citations in the markdown document @mjtooley ? Maybe @feamster has recommendations based on his experience. Most of mine is working not in markdown but in raw XML so not really applicable here.

Some options:
https://stackoverflow.com/questions/26587527/cite-a-paper-using-github-markdown-syntax
https://blog.martinfenner.org/2013/06/19/citations-in-scholarly-markdown/
https://v4.chriskrycho.com/2015/academic-markdown-and-citations.html
https://ulyngs.github.io/oxforddown/cites-and-refs.html
https://gist.github.com/maxogden/97190db73ac19fc6c1d9beee1a6e4fc8
https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html

Also, a general reference for newbies: https://guides.github.com/features/mastering-markdown/

**Comment by mjtooley** [comment]
I think we need Nick's input here.  From my little bit of research on this it looks like most use run the markdown file through Pandoc that cross-references the citation placeholders with the bibliography file to generate a new md file.  Or at least that is what I think they say to do.

For us, I think we need to think about the work flow as the generation is at the end when getting ready to publish the document.  I think we may want to come up a with simple method for marking the citations in the paper (e.g. [1] or [Feamster2019] and then keeping a master citation list either at the end of the draft document or in a separate file.

**Comment by bhstark2** [comment]
I can provide input. I use md for docs in both IETF and BBF. Both do xrefs. There are decisions to make as to whether we want to ultimately run the md through a processor and create html, or make direct use of the md.

**Comment by jlivingood** [comment]
Did we come to a decision on how to do xrefs? What do you suggest @bhstark2?

**Comment by bhstark2** [comment]
I'd like to suggest that we use pandoc to generate html5 and pdf versions of the doc for publication. This is one of the options mentioned in the links @jlivingood listed. I've had really good experience with it. We'll have a separate file where the references are put, in yaml format (which is really easy to use). Each reference will have an id. The references will be referred to using the syntax `[@<id>]`.

Here is what a yaml reference might look like (in the separate reference file):
`
- id: RFC5246
  call-number: RFC 5246
  title: The Transport Layer Security (TLS) Protocol Version 1.2
  published: IETF
  issued: {year: 2008}
  URL: https://tools.ietf.org/html/rfc5246
`
There are a lot more ref metadata fields that are defined and can be used. And then a ref to this from the main doc would be `[@RFC5246]`. 
I'll work on figuring out more pandoc conventions and how to do the conversion. I haven't actually done it myself, but I know people who have and I love the results. They're also really happy with how easy it is.

**Comment by feamster** [comment]
+1 to pandoc, which I (independently!) just mentioned in the other thread. I've had very good luck going between various formats to github markdown (from HTML, dokuwiki, docx, etc.).  I've never tried it going the other way, and I've never tried the YAML references, but this seems totally cool!

I should read more carefully, but I assume there's some way to cross-ref sections and figures in Markdown?  That would be the last big gap in my own knowledge, I think.

**Comment by bhstark2** [comment]
There are ways to do section xrefs, especially if html (and subsequently PDF) is the goal. Among different md varieties, not so much. If going -> html -> PDF, the rules go easiest into the css created for the html version. The css is simply one of the artifacts that hangs out in the repo, and gets used by pandoc. In my other project, we use a convention of (@sec:<section-name>) to do a reference to a section. The anchors are auto-created where a header titled "This Section Header" is referenced as "(@sec:this-section-header). But it's also possible to provide explicit anchors like "[#sec:no-i-want-it-this-way]". This is useful if there are multiple sections with the same heading title.

**Comment by jlivingood** [comment]
I think you are right on HTML & PDF as the intended output, so then it sounds as if section xrefs are doable! :-)

**Comment by jlivingood** [comment]
Barbara - would this example of usage be correct based on your directions?

https://github.com/bitag-twg-org/bitag2.0/blob/master/pandemic_performance/HowToMarkdown.md

**Comment by bhstark2** [comment]
I think explicitly anchoring sections is only needed when we don't want to use the default, or if there are multiple sections with the same name. So, more like:

Cross-referencing sections:

Default anchor for a section heading will be `[#sec:<heading title in lower case with all non-alphanumeric characters replaced by hyphen "-">]`. If this default is acceptable , no need to create the anchor explicitly. If something other than the default is desired, define it explicitly. Example 1:
'# Introduction`
To refer back to that section then:
'(@sec:introduction)'
Example 2:
'# Introduction [#sec:intro]'
'(@sec:intro)'

Example of xref in context:
'As we noted in (@sec:introduction) it is important to note that...'

**Comment by jlivingood** [comment]
Makes sense! Updated https://github.com/bitag-twg-org/bitag2.0/blob/master/pandemic_performance/HowToMarkdown.md accordingly.
